### PR TITLE
use wordsplit with taggers

### LINKED ISSUE #1929: [Bug] It fails when using wordsplit with pos_tag or ner_tag.
**Describe the bug**
When I train bidaf with pos_tag or ner_tag, it fails to begin training.

**To Reproduce**
Steps to reproduce the behavior
1. Copy `training_config/bidaf.jsonnet` to custom directory.
2. Modify `bidaf.jsonnet` as follow:
a) change `toke_indexers` of `dataset_reader`
```jsonnet
    "token_indexers": {
      // config `tokens` and `token_characters` as before ...
      "pos_tag": {
        "type": "pos_tag"
      },
      "ner_tag": {
        "type": "ner_tag"
      }
    }
```
b) change `text_field_embedder` of `model`:
```jsonnet
    "text_field_embedder": {
       // config token_embedders here ...
      "allow_unmatched_keys": true
    },
```
3. apply command: `allennlp train config/bidaf.jsonnet -s data/bidaf`
4. See error as follow:
```bash
KeyError: '@@UNKNOWN@@'
```
or
```bash
RuntimeError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
```

**Expected behavior**
No error and same behaviour as before.

**System (please complete the following information):**
 - OS: Linux
 - Python version: 3.6.1
 - AllenNLP version: 0.6.1
 - PyTorch version: 0.4.0

**Additional context**
I find there are three code snippets that may be related to this bug. I am trying to fix it.
- [allennlp/data/token_indexers/ner_tag_indexer.py](https://github.com/allenai/allennlp/blob/master/allennlp/data/token_indexers/ner_tag_indexer.py#L41)
- [allennlp/data/token_indexers/pos_tag_indexer.py](https://github.com/allenai/allennlp/blob/master/allennlp/data/token_indexers/pos_tag_indexer.py#L59)
- [allennlp/nn/util.py](https://github.com/allenai/allennlp/blob/master/allennlp/nn/util.py#L428)

### PR DESCRIPTION
**Pull request by WrRan**
Fixes #1929.

This PR is same as #1935.

I have deleted the original repo, which makes #1935 from `unknown repository`.
And I do not how to solve this problem.
So I create this PR.

### CODE REVIEW

**Comment by WrRan** [comment]
Though PR passes all test-cases, yet I feel uneasy to modify previous test cases related to this issue.

File: allennlp/tests/data/token_indexers/ner_tag_indexer_test.py [hunk 1]
```diff
@@ -41,3 +41,13 @@ def test_as_array_produces_token_sequence(self):
...
+
+    def test_blank_ner_tag(self):
+        tokens = self.tokenizer.split_words("allennlp is awesome.")
+        indexer = NerTagIndexer()
+        counter = defaultdict(lambda: defaultdict(int))
+        for token in tokens:
+            indexer.count_vocab_items(token, counter)
+        vocab = Vocabulary(counter)
+        # should raise no exception
+        indexer.tokens_to_indices(tokens, vocab, index_name="ner")
```
Comment (matt-gardner) [inline allennlp/tests/data/token_indexers/ner_tag_indexer_test.py]: It's not obvious in this test that the NER tags are blank.  Can you `assert` something about that in here?  Or you could just construct `Token` objects yourself that have some NER tags with values, and some with blanks.  Also, it'd be good to have an actual assert on the result of this line, that you're getting the value for "NONE" where you expect.

File: allennlp/tests/data/token_indexers/pos_tag_indexer_test.py [hunk 2]
```diff
@@ -59,3 +59,13 @@ def test_as_array_produces_token_sequence(self):
...
+
+    def test_blank_pos_tag(self):
+        tokens = self.tokenizer.split_words("allennlp is awesome.")
+        indexer = PosTagIndexer()
+        counter = defaultdict(lambda: defaultdict(int))
+        for token in tokens:
+            indexer.count_vocab_items(token, counter)
+        vocab = Vocabulary(counter)
+        # should raise no exception
+        indexer.tokens_to_indices(tokens, vocab, index_name="pos")
```
Comment (matt-gardner) [inline allennlp/tests/data/token_indexers/pos_tag_indexer_test.py]: This one doesn't actually test a blank pos tag, does it?  Maybe add a blank pos tag, and check the result?

**Comment by schmmd** [comment]
@matt-gardner approved?

**Pull request merged**